packages/procaptcha/src/modules/Manager.ts

```diff
@@ -597,12 +596,34 @@ export function Manager(
     }
 
     const exportData = async (events: StoredEvents) => {
+        const providerUrlFromStorage = storage.getProviderUrl()
+        let providerApi: ProviderApi
+
+        if (providerUrlFromStorage) {
+            providerApi = await loadProviderApi(providerUrlFromStorage)
+        } else {
+            const contract = await loadContract()
+            const getRandomProviderResponse: RandomProvider = await wrapQuery(
+                contract.query.getRandomActiveProvider,
+                contract.query
+            )(getAccount().account.address, getDappAccount())
+            const providerUrl = trimProviderUrl(getRandomProviderResponse.provider.url.toString())
+            providerApi = await loadProviderApi(providerUrl)
+        }
+
         const providerUrl = storage.getProviderUrl() || state.captchaApi?.provider.provider.url.toString()
         if (!providerUrl) {
             return
         }
-        const providerApi = await loadProviderApi(providerUrl)
-        await providerApi.submitUserEvents(events, getAccount().account.address)
+        console.log('Submitting events to provider', events)
+
+        let account = ''
+        try {
+            account = getAccount().account.address
+        } catch (e) {
+            console.error(e)
```

Comment:
this will also proceed onwards with address as '', which means we'll have several db records with no account if something went wrong here. Not sure if that's a problem, just flagging it

Reply:
Yeah agreed - more interested in getting data into the db when getAccount() fails (which can happen in a dev environment)

Reply:
sure, makes sense :)